llvm/lib/Transforms/Vectorize/VPlanValue.h

```diff
@@ -82,12 +70,15 @@ class LLVM_ABI_FOR_TEST VPValue {
   /// Return the underlying Value attached to this VPValue.
   Value *getUnderlyingValue() const { return UnderlyingVal; }
 
+  /// Return the underlying IR value for a VPIRValue.
+  Value *getLiveInIRValue() const;
+
   /// An enumeration for keeping track of the concrete subclass of VPValue that
   /// are actually instantiated.
   enum {
-    VPValueSC, /// A generic VPValue, like live-in values or defined by a recipe
-               /// that defines multiple values.
-    VPVRecipeSC /// A VPValue sub-class that is a VPRecipeBase.
+    VPVLiveInSC,   /// A live-in VPValue wrapping an IR Value.
+    VPVSymbolicSC, /// A symbolic live-in VPValue without IR backing.
+    VPVDefValueSC, /// A VPValue defined by a recipe.
```

Comment:
```suggestion
    VPVRecipeValueSC, /// A VPValue defined by a recipe.
```
or use DefValue consistently.